R/art.con.R

```diff
@@ -0,0 +1,125 @@
+#'   compared. See "Formula" section below.
+#' @param response Which response to use: the aligned (with ART-C) response
+#'   (\code{"aligned"}) or the aligned-and-ranked (with ART-C) response
+#'   (\code{"art"}). Default is "art". This argument is passed to artlm.con.
+#' @param factor.contrasts The name of the contrast-generating function to be
+#'   applied by default to fixed effect factors. Sets the the first element of
+#'   \code{\link{options}("contrasts")} for the duration of this function. The
+#'   default is to use \code{"contr.sum"}, i.e. sum-to-zero contrasts, which is
+#'   appropriate for Type III ANOVAs (the default ANOVA type for
+#'   \code{\link{anova.art}}). This argument is passed to artlm.con.
+#' @param method Contrast method argument passed to \code{\link{contrast}}.
+#'   Note: default is \code{"pairwise"} but \code{\link{contrast}} default is
+#'   \code{"eff"}.
+#' @param interaction Logical value. Default is FALSE. If TRUE, conducts
+#'   interactions contrasts directly on \code{\link{art}} model m (i.e., does
+#'   not align and rank with ART-C procedure). See “Interaction contrasts”
+#'   section in \code{\link{contrast}}.
+#' @param adjust Character: adjustment method (e.g., "bonferroni") – passed to
+#'   \code{\link{contrast}}. If not provided, \code{\link{contrast}} will use
+#'   its default ("tukey" at the time of publication). All available options are listed
+#'   in \code{\link{summary.emmGrid}} in \code{P-value adjustments} section.
+#' @param \dots Additional arguments passed to \code{\link{lm}} or
+#'   \code{\link{lmer}}.
+#' @return An object of class \code{emmGrid}. See \code{\link{contrast}}
+#'   for details.
+#' @author Lisa A. Elkin
+#'
+#' @section Formula: Contrasts compare combinations of levels from multiple
+#'   factors. The \code{f} parameter indicates which factors are involved. Two
+#'   formats are accepted: a \code{"character"} term as used in
+#'   \code{\link{artlm}} and \code{\link{artlm.con}}, or a \code{"formula"} as
+#'   used in \code{\link{emmeans}}. For example, contrasts comparing
+#'   combinations of levels of factors \emph{X1} and \emph{X2} can be expressed
+#'   as "X1:X2" (\code{"character"} term) or as ~ X1*X2 (\code{"formula"}).
+#'
+#' @references Lisa A. Elkin, Matthew Kay, James J. Higgins, Jacob O. Wobbrock.
+#'   Under review.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' data(Higgins1990Table5, package="ARTool")
+#'
+#' ## Perform aligned rank transform
+#' m <- art(DryMatter ~ Moisture*Fertilizer + (1|Tray), data=Higgins1990Table5)
+#'
+#' ## In a typical workflow, contrast tests using ART-C would follow a
+#' ## significant omnibus effect found by running an anova on the ART responses
+#' ## (equivalent to anova(m, response="art")).
+#' ## If conducting planned contrasts, this step can be skipped.
+#' anova(m)
+#'
+#' ## We can conduct contrasts comparing levels of Moisture using the ART-C procedure.
```

Comment:
I'd rather avoid all the post hoc / omnibus style language in the package docs.... it's not an approach i want to advocate and if people are already doing it they can figure things out from us showing stuff on contrasts